nav2_dwb_controller/dwb_controller/src/dwb_controller.cpp

```diff
@@ -124,77 +123,112 @@ DwbController::on_shutdown(const rclcpp_lifecycle::State &)
   return nav2_util::CallbackReturn::SUCCESS;
 }
 
-void
-DwbController::followPath(const std::shared_ptr<GoalHandle> goal_handle)
+void DwbController::followPath()
 {
   RCLCPP_INFO(get_logger(), "Received a goal, begin following path");
-  auto result = std::make_shared<nav2_msgs::action::FollowPath::Result>();
-
-  std::shared_ptr<GoalHandle> current_goal_handle = goal_handle;
-
-  rclcpp::Rate loop_rate(100ms);    // period vs. hz
-
-  auto goal = current_goal_handle->get_goal();
 
   try {
-    auto path = nav_2d_utils::pathToPath2D(goal->path);
-
-    RCLCPP_DEBUG(get_logger(), "Providing path to the local planner");
-    planner_->setPlan(path);
-
+    setPlannerPath(action_server_->get_current_goal()->path);
     ProgressChecker progress_checker(rclcpp_node_);
 
-    rclcpp::Rate loop_rate(10);
+    rclcpp::Rate loop_rate(100ms);
     while (rclcpp::ok()) {
-      nav_2d_msgs::msg::Pose2DStamped pose2d;
-      if (!getRobotPose(pose2d)) {
-        RCLCPP_INFO(get_logger(), "No pose. Stopping the robot");
+      if (shouldCancel()) {
```

Comment:
This does a lot more than just check if we should cancel. It changes state. I feel like the code should read more like

```
 while(rclcpp::ok()) {
  update_goal(goal_handle);
  if (goal_handle.empty())
    return;
```

update_goal would then handle the cancelling and preemption.

Reply:
I agree that it's a bit unexpected. Reading the code, I though shouldCancel would only tell me if I should cancel or not. However, it is returning if shouldCancel is true. However, the code doesn't do anything in response to knowing that it should cancel. Recommend reworking this section to clarify.

Reply:
Agree, I'll push a change.